client/disputes/evidence/index.js

```diff
@@ -367,7 +367,7 @@ export default ( { query } ) => {
 			}
 		);
 
-	useConfirmNavigation( () => {
```

Comment:
See `client/utils/use-confirm-navigation.js` for explanation on the slight refactor.

The effect is still called when the dependencies change.

So instead of:
```
const useCustomHook = (callback, dependencies) => {
  useEffect(() => {callback()}, dependencies);
}
...
useCustomHook(() => thing(), [dependency]);
```

Now we have:
```
const useCustomHook = (callback) => callback;
...
const callback = useCustomHook(() => thing());
useEffect(() => callback(), [callback, dependency]);
```

Same difference - but it makes it more explicit which dependencies need to run and it now has a stable reference on the callback.